pulpcore/cli/core/role.py

```diff
@@ -0,0 +1,80 @@
+import gettext
+from typing import Iterable, Optional
+
+import click
+
+from pulpcore.cli.common.context import PluginRequirement, PulpContext, pass_pulp_context
+from pulpcore.cli.common.generic import (
+    create_command,
+    destroy_command,
+    href_option,
+    list_command,
+    name_option,
+    show_command,
+    update_command,
+)
+from pulpcore.cli.core.context import PulpRoleContext
+
+_ = gettext.gettext
+NO_PERMISSION_KEY = "pulpcore.cli.core.role.no_permission"
+
+
+def _no_permission_callback(ctx: click.Context, param: click.Parameter, value: bool) -> bool:
+    ctx.meta[NO_PERMISSION_KEY] = value
+    return value
+
+
+def _permission_callback(
+    ctx: click.Context, param: click.Parameter, value: Iterable[str]
+) -> Optional[Iterable[str]]:
+    if ctx.meta.get(NO_PERMISSION_KEY, False):
+        if value:
+            raise click.ClickException(_("Cannot specify `--permission` and `--no-permission`."))
+        return []
+    return value or None
+
+
+filters = [
+    click.option("--locked/--unlocked", default=None),
+    click.option("--name"),
+    click.option("--name-in", "name__in"),
+    click.option("--name-contains", "name__contains"),
+    click.option("--name-icontains", "name__icontains"),
+    click.option("--name-startswith", "name__startswith"),
```

Comment:
Lots of options here as well. It's fine if you think we should have them all, I'm just unsure of how useful they all are.

Reply:
`--name-startswith "core."` is probably helpful, as is `--name-contains "fileremote"`. About the `icontains` i'm not so sure.